Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,65 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith(() =>
+                {
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+                    string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+                    return new FailReason($"Expected enum to equal {expectationName} by value{{reason}}, but found {subjectsName}.");
+                });
+        }
+
+        private static void HandleByName(IEquivalencyValidationContext context)
+        {
+            string subject = context.Subject.ToString();
+            string expected = context.Expectation.ToString();
+
+            Execute.Assertion
+                .ForCondition(subject == expected)
+                .FailWith(() =>
+                {
+                    decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+                    decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+                    string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+                    string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+                    return new FailReason(
+                            $"Expected enum to equal {expectationName} by name{{reason}}, but found {subjectsName}.");
+                });
+        }
+
+        private static string EnumDescription(object o, decimal? v)
+        {
+            string PrintDecimal(decimal x) => x.ToString(CultureInfo.InvariantCulture);
+
+            if (o == null || v == null)
+            {
+                return "null";
+            }
+
+            if (o.GetType().GetTypeInfo().IsEnum)
+            {
+                string typePart = o.GetType().Name;
+                string namePart = Enum.GetName(o.GetType(), o);
+                string valuePart = PrintDecimal(v.Value) ?? "?";
```

Comment:
The `?? "?"` should not be necessary since the redefinition of `PrintDecimal`

Reply:
Removed, thanks.